Added security spec to jcasc values

miles-w-3 opened the pull request:
Signed-off-by: Miles Wilson <wilson.mil@icloud.com>

<!--
Thank you for contributing!
Before you submit this PR we'd like to make sure you are aware of our technical requirements and best practices:

* https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#technical-requirements
* https://helm.sh/docs/chart_best_practices/

For a quick overview across what we will look at reviewing your PR, please read our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the history.
This will make it easier to identify new changes.
The PR will be squashed anyways when it is merged.
Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them.
Once pushed, GitHub Actions will run across your changes and do some initial checks and linting.
These checks run very quickly.
Please check the results.
We would like these checks to pass before we even continue reviewing your changes.
-->
<!-- markdownlint-disable MD041 -->

### What this PR does / why we need it

### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #684 

### Special notes for your reviewer


### Checklist
<!-- [Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->
- [X] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [X] CHANGELOG.md was updated

miles-w-3 commented:
had to add some extensive if logic to include the original security settings in a variety of cases. Seems like this is an issue in helm `3.8.2` but not `3.9`, which is interesting

timja reviewed (commented):
can you add some unit tests please?

see https://github.com/jenkinsci/helm-charts/tree/main/charts/jenkins/unittests

and the GitHub action for how to run

miles-w-3 commented:
Simplified logic to let the apiToken default live in the `values.yaml` file. Added info about the new field to the values file as well as the values summary readme.
Adding unit tests for two cases:
1. The security section is added to, but apiToken is not changed from the default
2. apiToken is modified in downstream values using the security section
